smarts/core/agents_provider.py

```diff
@@ -222,7 +222,7 @@ def __init__(self, sim):
         super().__init__(sim)
 
     @property
-    def action_spaces(self) -> Set[ActionSpaceType]:
+    def actions(self) -> Set[ActionSpaceType]:
```

Comment:
Why did we rename `action_spaces` to `actions`?

Reply:
This was to bring it in line with the agent interface which had deprecated `action_space` for `action`.

Reply:
Truthfully, I want to rename `ActionSpaceType` to `ActionType` because we are dealing with shaped actions directly without regard to limitations.